spec/consumer_spec.rb

```diff
@@ -21,6 +22,33 @@ def consume(_payload, _metadata)
         end
       end
       stub_const('ConsumerTest::MyConsumer', consumer_class)
+      schema_class = Class.new(Deimos::SchemaClass::Record) do
+        def schema
+          'MySchema'
+        end
+
+        def namespace
+          'com.my-namespace'
+        end
+
+        attr_accessor :test_id
+        attr_accessor :some_int
+
+        def initialize(test_id: nil,
+                       some_int: nil)
+          self.test_id = test_id
+          self.some_int = some_int
+        end
+
+        def as_json(_opts={})
+          {
+            'test_id' => @test_id,
+            'some_int' => @some_int,
+            'payload_key' => @payload_key&.as_json
+          }
+        end
+      end
+      stub_const('Schemas::MySchema', schema_class)
```

Comment:
Is there a reason we have to add this in all four specs? If we're just trying to test overriding, can we have a dedicated spec for that? Also, why does the overriding class have to reimplement every method?

Reply:
`Schemas::MySchema` wasn't defined anywhere. So in tests where `SCHEMAS_CLASS_SETTING = on` test were failing.

Reply:
It should be in `spec/schemas/my_namespace`. If it's not showing up, maybe the settings are wrong for the tests? Or do we need to add it to `spec/schemas` and require it similarly to the other generated schema files?